python/ray/tune/config_parser.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+import argparse
+import json
+import numpy as np
+import os
+import random
+import sys
+
+from ray.tune.trial import Trial
+
+
+def make_parser(description):
+    """Returns a base argument parser for the ray.tune tool."""
+
+    parser = argparse.ArgumentParser(description=(description))
+
+    parser.add_argument("--alg", default="PPO", type=str,
+                        help="The learning algorithm to train.")
+    parser.add_argument("--stop", default="{}", type=json.loads,
+                        help="The stopping criteria, specified in JSON.")
+    parser.add_argument("--config", default="{}", type=json.loads,
+                        help="The config of the algorithm, specified in JSON.")
+    parser.add_argument("--resources", default='{"cpu": 1}', type=json.loads,
+                        help="Amount of resources to allocate per trial.")
+    parser.add_argument("--num_trials", default=1, type=int,
+                        help="Number of trials to evaluate.")
+    parser.add_argument("--local_dir", default="/tmp/ray", type=str,
+                        help="Local dir to save training results to.")
+    parser.add_argument("--upload_dir", default=None, type=str,
+                        help="URI to upload training results to.")
+    parser.add_argument("--checkpoint_freq", default=sys.maxsize, type=int,
+                        help="How many iterations between checkpoints.")
+
+    # TODO(ekl) environments are RL specific
+    parser.add_argument("--env", default=None, type=str,
+                        help="The gym environment to use.")
+
+    return parser
+
+
+def parse_to_trials(config):
+    """Parses a json config to the number of trials specified by the config.
+
+    The input config is a mapping from experiment names to an argument
+    dictionary describing a set of trials. These args include the parser args
+    documented in make_parser().
+
+    Additionally, variables in the `config` section may be set to different
+    values for each trial. You can either specify `grid_search: <list>` in
+    place of a concrete value to specify a grid search across the list of
+    values, or `eval: <str>` for values to be sampled from the given Python
+    expression.
+
+    See ray/rllib/tuned_examples for more examples of configs in YAML form.
```

Comment:
I feel we will easily lose track of this line (especially as we move the example code around), might make sense to move this to a README

Reply:
along with the example yaml